src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java

```diff
@@ -320,8 +378,13 @@ public boolean checkUserPermission(ResponseEntity responseEntity, UUID documentI
                 .getCaseDocumentMetadata(caseId, documentId, authorization)
                 .orElseThrow(() -> new CaseNotFoundException(caseId));
 
-            return (caseDocumentMetadata.getDocument().getId().equals(documentId.toString())
-                    && caseDocumentMetadata.getDocument().getPermissions().contains(Permission.READ));
+            List<Document> matchingDocuments =
+                caseDocumentMetadata.getDocuments().stream()
+                                    .filter(document -> document.getId().equals(documentId.toString()))
+                                    .filter(document -> document.getPermissions().contains(Permission.READ))
+                                    .collect(Collectors.toList());
+
+            return matchingDocuments.size() > 0;
         }
```

Comment:
I think, this block won't be required if we keep the caseDocumentMetadata update with single document.